copy-edit-topic-agile-kanban-wastes (#556)
Umuzi-org/ACN-syllabus: content/agile/kanban-wastes/_index.md

You can think of kanban as a way to prioritize work. Kanban is a methodology used in professional dev teams and even manufacturing. 

We use Kanban techniques because:
- it's common in the industry, so junior coders must get it.
- it works well, we use kanban to keep people moving in the right direction and make sure everyone has the support that they need

## A good day

When we are getting things right, then the following should all be true:

> 1. students should be working on their highest priority task at all times.

Reviewer: ```suggestion
1. Students should be working on their highest priority task at all times.
```

> 2. students should be working at a sustainable rate, they shouldn't burn themselves out or go too slow.

Reviewer: ```suggestion
2. Students should be working at a sustainable rate, they shouldn't burn themselves out or go too slow.
```

> 3. students should learn the good habits that make work on a professional dev team successful.

Reviewer: ```suggestion
3. Students should learn the good habits that make work on a professional dev team successful.
```

> 4. if there are any problems then those problems need to be surfaced and dealt with quickly. The longer a problem waits to be addressed, the more costly it is to fix.

Reviewer: ```suggestion
4. If there are any problems then those problems need to be surfaced and dealt with quickly. The longer a problem waits to be addressed, the more costly it is to fix.
```

5. Everyone gets their work done in a way that grows their understanding and skills.

## Sources of waste

Kanban is a method for eliminating waste. It is a methodology that comes from manufacturing, but it makes a lot of sense in the context of software development as well.

What is waste? There are a few different kinds:

### Partially done work

This is somewhere where resources have been sunk, but no value has yet been created.

> Eg: imagine you owned a car factory. You can't sell half a car. You can only sell a car that is complete. Same with software. Half a feature is useless. Completing work adds value.

Reviewer: ```suggestion
For example: imagine you owned a car factory. You can't sell half a car. You can only sell a car that is complete. Same with software. Half a feature is useless. Completing work adds value.
```
